### ISSUE TITLE
Can the program list (all) pairwise distinct longest cycles ?

### ISSUE THREAD

**Issue by lichengzhang1** [JorikJooken/longestCycles#1]
This program is excellent, but sometimes we not only need counting but also want to list some or all  longest cycles. Can this program provide such an output? By the way, there's a sentence in the [README.md](https://github.com/JorikJooken/longestCycles/blob/master/README.md) that confuses me: `This indicates that the graph is non-hamiltonian and has 4 pairwise distinct hamiltonian cycles.`

**Comment by JorikJooken** [comment]
Thank you for your message! I corrected the sentence in the README file (I meant "longest cycles" instead of "hamiltonian cycles").

Moreover, the algorithm described in ""Improved asymptotic upper bounds for the minimum number of pairwise distinct longest cycles in regular graphs" can also be used to list all longest cycles. For your convenience, I have just added a program "enumerateLongestCycles.cpp" that supports this function and I have updated the README to indicate how to use the program.
